core/utils.py

```diff
@@ -208,44 +208,90 @@ def spread_default_parameters(config, dev_cfg):
 
-def verify_connections_layout(comm, log, name, triggers=None):
-    """checks if the subdictionaries in the connections section
-    are valid triggers
-
-    comm: the communications dictionary will all connections
-    log: the log instance of the device
-    name: the name of the device
-    triggers: a list of valid values for device triggers
+def verify_connections_layout(comm, log, name, outputs=None):
+    """
+    Use this method at the end of the sensor initialisation
+    before calling configure_device_channel().
+    Checks the YAML configuration to make sure the subdictionaries
+    in the connections section are valid outputs
+
+    comm:    the communications dictionary with all connections
+    log:     the log instance of the device
+    name:    the name of the device
+    outputs: optional, a list of valid values for device outputs.
+             If omitted, the function will ensure that the YAML
+             configuration does not specify any output channels for the device.
+             Expects the output_names used by a sensor as list e. g.:
+             outputs = [output_name1, output_name2]
     """
     for conn in comm.values():
+        #loop thru all connections
         if isinstance(conn, dict):
             for (key, value) in conn.items():
+                #loop thru sub items of the connections
+                #if sub item is a dict we found a output channel
                 if isinstance(value, dict):
-                    if not key in triggers:
-                        log.warning("%s has unknown outputs '%s' in Connections."
-                                    ' Valid outputs are: %s', name, key, triggers)
+                    if isinstance(outputs, list):
+                        if not key in outputs:
+                            log.warning("%s has unknown outputs '%s' in Connections."
+                                        ' Valid outputs are: %s', name, key, outputs)
+                    else:
+                        #handle case where outpus is not specified
+                        log.warning("%s has unexpected outputs '%s' in Connections."
+                                    ' No outputs are allowed', name, key)
 
 def configure_device_channel(comm:dict, *, is_output:bool,
                                 output_name:str = None, datatype:ChanType = ChanType.STRING,
                                 unit:str = None, name:str = None,
                                 restrictions:str = None):
-    """this method will set default values inside the connections section
-    so a connector which supports auto discover can register the device properly
+    """
+    Use this method at the end of the sensor/actuator initialisation,
+    it sets default values inside the connections section
+    so that a connector which supports auto-discover, e. g. homie_conn,
+    can register the device correctly.
+
+    Call this method once for each output the sensor has once, to register all output channel.
```

Comment:
```suggestion
    Call this method once for each output the sensor has, to register all output channel.
```

Not sure about this recommendation, however sounds awkward to have `once` two times.